src/network/src/esp/src/NetworkBroadcast.cpp

```diff
@@ -0,0 +1,103 @@
+#include "NetworkBroadcast.h"
+#include "NetworkConfig.h"
+#include "SocketFactory.h"
+#include "Task.h"
+#include "lwip/sockets.h"
+
+static void task(void* context) {
+    while (true) {
```

Comment:
Your task will never exit, even after the destructor is called, this could cause a segfault. Maybe you could use `m_started` or another variable?